gears/src/cli/query.rs

```diff
@@ -1,33 +1,41 @@
-use crate::{commands::client::query::QueryCommand, config::DEFAULT_TENDERMINT_RPC_ADDRESS};
+use crate::{
+    application::ApplicationInfo, cli::config::client_config, commands::client::query::QueryCommand,
+};
 use clap::{ArgAction, Subcommand, ValueHint};
+use std::marker::PhantomData;
 use tendermint::types::proto::block::Height;
 
 /// Querying subcommands
 #[derive(Debug, Clone, ::clap::Args)]
-pub struct CliQueryCommand<C: Subcommand> {
+pub struct CliQueryCommand<T: ApplicationInfo, C: Subcommand> {
     /// <host>:<port> to Tendermint RPC interface for this chain
```

Comment:
I think we should clarify the behaviour here e.g.
`Overrides any node in the client config. Default value is used if neither this argument nor a config value is provided`
+ Does the env variable override all of them.

Reply:
Added doc